File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <unordered_set>
using namespace std;

int solution(vector<int> nums)
{
    unordered_set<int> s(nums.begin(), nums.end());
    int answer = s.size();
    int l = nums.size() / 2;
    if (answer > l) {
        return l;
    }
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int n = nums.length;
        Set<Integer> s = new HashSet<>();
        for(int num: nums) {
            s.add(num);
        }
        if (s.size() >= n / 2) return n / 2;
        else return s.size();
    }
}